### PR TITLE
[Feature][Vectorized] support lateral view

### LINKED ISSUE #8447: [Feature][Vectorized] support lateral view
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

support vectorized lateral view.

### Use case

```sql
MySQL [test]> select * from test_lv_str lateral view explode_split(k2, ',') tmp1 as e1;
+------+-------+------+
| k1   | k2    | e1   |
+------+-------+------+
|    1 | a,b,c | a    |
|    1 | a,b,c | b    |
|    1 | a,b,c | c    |
+------+-------+------+
MySQL [test]> select * from test_lv_str lateral view explode_split(k2, ',') tmp1 as e1 lateral view explode_split(k2, ',') tmp2 as e2;
+------+-------+------+------+
| k1   | k2    | e2   | e1   |
+------+-------+------+------+
|    1 | a,b,c | a    | a    |
|    1 | a,b,c | a    | b    |
|    1 | a,b,c | a    | c    |
|    1 | a,b,c | b    | a    |
|    1 | a,b,c | b    | b    |
|    1 | a,b,c | b    | c    |
|    1 | a,b,c | c    | a    |
|    1 | a,b,c | c    | b    |
|    1 | a,b,c | c    | c    |
+------+-------+------+------+
```

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

Issue Number: close #8447

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/vrepeat_node.cpp [hunk 1]
```diff
@@ -181,13 +187,9 @@ Status VRepeatNode::get_next(RuntimeState* state, Block* block, bool* eos) {
 
     // current child block has finished its repeat, get child's next block
     if (_child_block->rows() == 0) {
-        if (_child_eos) {
```
[the remark this answers is not in the document]
Comment (BiteTheDDDDt) [reply be/src/vec/exec/vrepeat_node.cpp:184]: This is unnessesary code.

File: be/src/vec/exprs/table_function/vexplode_split.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    _delimiter_column = _delimiter_column->convert_to_full_column_if_const();
+
+    return Status::OK();
+}
+
+Status VExplodeSplitTableFunction::process_row(size_t row_idx) {
+    _is_current_empty = false;
+    _eos = false;
+
+    if (_text_column->is_null_at(row_idx)) {
```
Comment (HappenLee) [inline be/src/vec/exprs/table_function/vexplode_split.cpp:59]: here may be should only use `get_data_at` to replace `is_null_at`

File: be/src/exprs/table_function/table_function_factory.cpp [hunk 3]
```diff
@@ -21,25 +21,30 @@
...
         *fn = pool->add(new ExplodeSplitTableFunction());
-        return Status::OK(); 
+        return Status::OK();
+    } else if (fn_name == "vexplode_split") {
+        *fn = pool->add(new VExplodeSplitTableFunction());
+        return Status::OK();
     } else if (fn_name == "explode_bitmap") {
         *fn = pool->add(new ExplodeBitmapTableFunction());
-        return Status::OK(); 
+        return Status::OK();
```
[the remark this answers is not in the document]
Comment (BiteTheDDDDt) [reply be/src/exprs/table_function/table_function_factory.cpp]: This pr will only support a example like explode_split. other functions will return a error.

File: be/src/exec/table_function_node.cpp [hunk 4]
```diff
@@ -266,22 +268,27 @@ Status TableFunctionNode::get_next(RuntimeState* state, RowBatch* row_batch, boo
...
+                                        dest_slot, parent_slot_desc->type(),
+                                        row_batch->tuple_data_pool());
                         tuple_ptr->set_not_null(parent_slot_desc->null_indicator_offset());
                     } else {
                         tuple_ptr->set_null(parent_slot_desc->null_indicator_offset());
                     }
                 }
                 parent_tuple_row->set_tuple(tuple_idx, tuple_ptr);
-                tuple_ptr = reinterpret_cast<Tuple*>(reinterpret_cast<uint8_t*>(tuple_ptr) + parent_tuple_desc->byte_size());
+                tuple_ptr = reinterpret_cast<Tuple*>(reinterpret_cast<uint8_t*>(tuple_ptr) +
```
Comment (EmmyMiao87) [inline be/src/exec/table_function_node.cpp:290]: Why do you modify so many code formatting?

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**